pkg/util/podstore/podstore.go

```diff
@@ -0,0 +1,194 @@
+// Copyright 2023 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package podstore
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+	"k8s.io/klog/v2"
+	"k8s.io/utils/clock"
+)
+
+const (
+	DeleteQueueName = "pod_delete"
+	PodIPIndex      = "PodIP"
+	DelayTime       = time.Minute * 5
+)
+
+type PodStore struct {
+	curPod       cache.Indexer
+	prevPod      cache.Indexer
+	podsToDelete workqueue.DelayingInterface
+	// Mapping pod.uuid to LocalClock
+	timestampMap map[string]*LocalClock
+	clock        clock.WithTicker
```

Comment:
it's not clear to me why you need `clock.WithTicker` instead of simply `clock.Clock`?

Reply:
The `NewDelayingQueueWithCustomClock` is taking  `clock.WithTicker` as parameter. And I see it is used in `ipsec_certificate_controller.go` and `allocator.go`. It was `clock.Clock` in client-go@v0.21, but it is changed to `clock.WithTicker` in client-go@v0.24

Reply:
makes sense if it is required by `NewDelayingQueueWithCustomClock`